src/components/Editor/Breakpoint.js

```diff
@@ -48,6 +53,10 @@ class Breakpoint extends Component {
     const sourceId = selectedSource.get("id");
     const line = toEditorLine(sourceId, breakpoint.location.line);
 
+    if (isEmptyLine(editor.codeMirror, line)) {
```

Comment:
we should probably have a data based approach... where emptyLines comes from the store

```js
isEmptyLine(emptyLines, line)
```

or

```js
this.props.isEmptyLine(line)
///

in the connect function

isEmptyLine: line => isEmptyLineInSource(getState(), line)

// in reducer

function isEmptyLineInSource(state, line) {
   const emptyLines = getEmptyLines(state, selectedSource)
   return emptyLines.includes(line);
}
```